cpp/tests/listener.cpp

```diff
@@ -187,7 +160,15 @@ TEST_F(ListenerTest, RaiseOnError)
     _worker->progress();
 
   listenerContainer->endpoint = nullptr;
-  _worker->progress();
+  for (size_t attempts = 0; attempts < MaxProgressAttempts; ++attempts) {
+    try {
+      _worker->progress();
+      ep->raiseOnError();
```

Comment:
So `raiseOnError` doesn't clear the error state and therefore catching here and then breaking means we can then expect an error when we call it after the loop I suppose.

Reply:
That's right. It's basically a way to report the status in the form of an exception. I'm not sure this is the best way to do it, but for now it feels ok, it could be changed though if we find some better way to handle that.